Application/Sources/Content/SectionViewController.swift

```diff
@@ -427,24 +427,30 @@ extension SectionViewController {
         SectionViewController(section: .configured(.watchLater))
     }
 
-    @objc static func mediasViewController(forDay day: SRGDay, channelUid: String?) -> SectionViewController & DailyMediasViewController {
-        if let channelUid {
+    @objc static func mediasViewController(forDay day: SRGDay, transmission: SRGTransmission, channelUid: String?) -> SectionViewController & DailyMediasViewController {
+        if transmission == .radio, let channelUid {
+            SectionViewController(section: .configured(.radioEpisodesForDay(day, channelUid: channelUid)))
+        } else if transmission == .radio, let channelUid = ApplicationConfiguration.shared.radioHomepageChannels.first?.uid {
+            // FIXME: Load all radio episodes by date, not only from the first channel.
             SectionViewController(section: .configured(.radioEpisodesForDay(day, channelUid: channelUid)))
         } else {
             SectionViewController(section: .configured(.tvEpisodesForDay(day)))
         }
     }
 
-    @objc static func showsViewController(forChannelUid channelUid: String?, initialSectionId: String?) -> SectionViewController {
-        if let channelUid {
+    @objc static func showsViewController(forTransmission transmission: SRGTransmission, channelUid: String?, initialSectionId: String?) -> SectionViewController {
+        if transmission == .radio, let channelUid {
+            SectionViewController(section: .configured(.radioAllShows(channelUid: channelUid)), initialSectionId: initialSectionId)
+        } else if transmission == .radio, let channelUid = ApplicationConfiguration.shared.radioHomepageChannels.first?.uid {
+            // FIXME: Load all radio A to Z shows, not only from the first channel.
             SectionViewController(section: .configured(.radioAllShows(channelUid: channelUid)), initialSectionId: initialSectionId)
         } else {
             SectionViewController(section: .configured(.tvAllShows), initialSectionId: initialSectionId)
         }
     }
 
-    @objc static func showsViewController(forChannelUid channelUid: String?) -> SectionViewController {
-        showsViewController(forChannelUid: channelUid, initialSectionId: nil)
+    @objc static func showsViewController(forTransmission transmission: SRGTransmission, channelUid: String?) -> SectionViewController {
```

Comment:
Would suggest not repeating `transmission` in the signature here since there is a proper type here (unlike situations where a string is all we have).

```suggestion
    @objc static func showsViewController(for transmission: SRGTransmission, channelUid: String?) -> SectionViewController {
```

Reply:
Nice input and good reason.

Fixed: https://github.com/SRGSSR/playsrg-apple/commit/50fef96320c6dc9980757da0b18ba2bf722a8924